src/utils/logger.py

```
@@ -16,200 +16,179 @@
+    _file_sink_ids[utility] = sink_id
+
+
+def shutdown_logging() -> None:
+    """Remove all Loguru sinks to flush queued messages. Safe to call multiple times."""
+    global _sinks_initialized, _console_sink_id, _file_sink_ids
+    try:
+        # Remove file sinks
+        for sid in list(_file_sink_ids.values()):
+            try:
+                _loguru_logger.remove(sid)
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)